tests/sentry/snuba/test_metrics.py

```diff
@@ -1,292 +1,341 @@
-from dataclasses import dataclass
 from datetime import datetime
 from unittest import mock
 
+import pytest
 import pytz
 from django.utils.datastructures import MultiValueDict
 from snuba_sdk import Column, Condition, Entity, Granularity, Limit, Offset, Op, Query
 
+from sentry.sentry_metrics.indexer.postgres import PGStringIndexer
 from sentry.snuba.metrics import (
     MAX_POINTS,
     QueryDefinition,
     SnubaQueryBuilder,
     SnubaResultConverter,
 )
-
-
-@dataclass
-class PseudoProject:
-    organization_id: int
-    id: int
-
+from sentry.testutils import TestCase
 
 MOCK_NOW = datetime(2021, 8, 25, 23, 59, tzinfo=pytz.utc)
 
+_STRINGS = (
+    "abnormal",
+    "crashed",
+    "environment",
+    "errored",
+    "healthy",
+    "production",
+    "release",
+    "session.duration",
+    "session.status",
+    "session",
+    "staging",
+    "user",
+    "init",
+    "session.error",
+)
 
-@mock.patch("sentry.snuba.sessions_v2.get_now", return_value=MOCK_NOW)
-@mock.patch("sentry.api.utils.timezone.now", return_value=MOCK_NOW)
-def test_build_snuba_query(mock_now, mock_now2):
 
-    # Your typical release health query querying everything
-    query_params = MultiValueDict(
-        {
-            "query": [
-                "release:staging"
-            ],  # weird release but we need a string exising in mock indexer
-            "groupBy": ["session.status", "environment"],
-            "field": [
-                "sum(session)",
-                "count_unique(user)",
-                "p95(session.duration)",
-            ],
-        }
-    )
-    query_definition = QueryDefinition(query_params)
-    snuba_queries = SnubaQueryBuilder(PseudoProject(1, 1), query_definition).get_snuba_queries()
+class MetricsSnubaTest(TestCase):
```

Comment:
Is this test meant to be an end to end test for the metrics product ?
It seems to me the old version was a unit test for the abstractions the built over query and query result. If that is the case, it does not seem it needs to depend on postgres to work.
Unit tests on small modules have a big value as they would tell you immediately what is broken if a regression appears without the need to bring in a lot of dependencies.  So if this was meant to be like that, I would not add the persistence aspect to it.

Reply:
@jjbayer you may be the best person to answer this question. I can leave the mock indexer for this test in place

Reply:
Please leave the mock in place to start with. Then we can sync with @jjbayer about this

Reply:
> It seems to me the old version was a unit test for the abstractions the built over query and query result.
Yes, this was the intention, although I never enforced the mock (the test simply assumed that the `indexer` service was configured to the mock indexer).